src/features/import-projects/helpers.ts

```diff
@@ -47,63 +47,96 @@ export const updateOrCreateProject = async (
 
   // Remove project if prelimResult is "Remove"
   if (prelimResult && project[prelimResult] === "Remove") {
-    // If project is imported, delete it from the database
-    if (existingProject?.imported) {
-      try {
-        await dataSource
-          .createQueryBuilder()
-          .delete()
-          .from(Project)
-          .where("id = :id", { id })
-          .execute();
+    if (existingProject) {
+      // Remove the current round from rfRounds
+      const updatedRfRounds = (existingProject.rfRounds || []).filter(
+        (rfr) => rfr !== rfRound
+      );
 
-        console.log(
-          `[${new Date().toISOString()}] - INFO: Project Deleted. Project ID: ${id}`
-        );
-      } catch (error) {
-        // make project as not imported if failed to delete because it may have some vouches
-        console.log(
-          `[${new Date().toISOString()}] - ERROR: Failed to delete project. Project ID: ${id}`
-        );
-        const updatedProject = new Project({
-          ...existingProject,
-          imported: false,
-        });
+      if (updatedRfRounds.length === 0) {
+        // If no rounds remain, delete the project
+        if (existingProject.imported) {
+          try {
+            await dataSource
+              .createQueryBuilder()
+              .delete()
+              .from(Project)
+              .where("id = :id", { id })
+              .execute();
+
+            console.log(
+              `[${new Date().toISOString()}] - INFO: Project Deleted. Project ID: ${id}`
+            );
+          } catch (error) {
+            // Mark project as not imported if failed to delete
+            console.log(
+              `[${new Date().toISOString()}] - ERROR: Failed to delete project. Project ID: ${id}, Error: ${error.message}`
+            );
+            try {
+              await dataSource
+                .createQueryBuilder()
+                .update(Project)
+                .set({ imported: false, rfRounds: updatedRfRounds })
+                .where("id = :id", { id })
+                .execute();
+
+              console.log(
+                `[${new Date().toISOString()}] - INFO: Project marked as not imported. Project ID: ${id}`
+              );
+            } catch (updateError) {
+              console.log(
+                `[${new Date().toISOString()}] - ERROR: Failed to mark project as not imported. Project ID: ${id}, Error: ${updateError.message}`
+              );
+            }
+          }
```

Comment:
Here it's assumed that having connected attestation is the only reason for deleting failure. It's not a 100% valid point and also reduces the readability